src/wrapper/objects/auto_byte_array.rs

```diff
@@ -58,17 +58,45 @@ impl<'a, 'b> AutoByteArray<'a, 'b> {
         self.ptr.as_ptr()
     }
 
-    /// Commits the result of the array, if it is a copy
+    fn commit_byte_array_elements(&mut self) -> Result<()> {
+        jni_void_call!(
+            self.env.get_native_interface(),
+            ReleaseByteArrayElements,
+            *self.obj,
+            self.ptr.as_mut(),
+            sys::JNI_COMMIT
+        );
+        Ok(())
+    }
+
+    fn release_byte_array_elements(&mut self) -> Result<()> {
+        jni_void_call!(
+            self.env.get_native_interface(),
+            ReleaseByteArrayElements,
+            *self.obj,
+            self.ptr.as_mut(),
+            self.mode as i32
+        );
+        Ok(())
+    }
+
+    /// Commits the changes to the array, if it is a copy
     pub fn commit(&mut self) {
-        let res = self
-            .env
-            .commit_byte_array_elements(*self.obj, unsafe { self.ptr.as_mut() });
+        let res = self.commit_byte_array_elements();
         match res {
             Ok(()) => {}
             Err(e) => debug!("error committing byte array: {:#?}", e),
```

Comment:
error, so that users are more likely to see and investigate it?

Reply:
Sure. I wanted to keep the API as similar as possible, but I think this makes sense.